### PR TITLE
fix: indexer validation

### PR DESCRIPTION
**Pull request by greged93**
Time spent on this PR: 0.2 day

Resolves: #1113 

# Pull Request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

# What is the new behavior?
Skips transactions which fail ethereum validation. Also adds deno test to the CI.

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: indexer/src/utils/filter.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+// Starknet
+import { Transaction, Event } from "../deps.ts";
+
+const KAKAROT_ADDRESS = Deno.env.get("KAKAROT_ADDRESS");
+if (KAKAROT_ADDRESS === undefined) {
+  throw new Error("ENV: KAKAROT_ADDRESS is not set");
+}
+
+export const isKakarotTransaction = (transaction: Transaction) => {
```
Comment (tcoratger) [inline indexer/src/utils/filter.ts:9]: I see there are unit tests written for the other `ethValidationFailed` function, maybe as a follow up we could have a good first issue with unit tests also for this function, wdyt?

File: indexer/src/utils/filter.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    console.error(JSON.stringify(transaction, null, 2));
+    return false;
+  }
+  // TODO(Greged93): replace this with a more robust check.
+  // ⚠️ The existence of `to` field in invoke calldata in RPC is not enforced by protocol.
+  // Forks or modifications of the kkrt-labs/kakarot-rpc codebase could break this check.
+  if (BigInt(to) !== BigInt(KAKAROT_ADDRESS!)) {
+    console.log("✅ Skipping transaction that is not related to Kakarot");
+    return false;
+  }
```
Comment (tcoratger) [inline indexer/src/utils/filter.ts:35]: Do we have an issue to track this?

File: indexer/src/utils/filter.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  return true;
+};
+
+export const ethValidationFailed = (event: Event) => {
```
Comment (tcoratger) [inline indexer/src/utils/filter.ts:39]: As you are using a lot the `data` field of `event` in this method, maybe you could have something like

```ts
  const { data } = event;
```

at the beginning of the function, doing then:

```ts
if (data.length === 0) {
    return false;
  }
```

File: indexer/src/utils/filter.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+export const isKakarotTransaction = (transaction: Transaction) => {
+  // Filter out transactions that are not related to Kakarot.
+  // callArrayLen <- calldata[0]
+  // to <- calldata[1]
+  // selector <- calldata[2];
+  // dataOffset <- calldata[3]
+  // dataLength <- calldata[4]
+  // calldataLen <- calldata[5]
+  const calldata = transaction.invokeV1?.calldata;
+  if (calldata === undefined) {
```
Comment (tcoratger) [inline indexer/src/utils/filter.ts:18]:
```suggestion
  if (!calldata) {
```

File: indexer/src/utils/filter.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    console.error("No calldata in transaction");
+    console.error(JSON.stringify(transaction, null, 2));
+    return false;
+  }
+  const to = calldata[1];
+  if (to === undefined) {
```
Comment (tcoratger) [inline indexer/src/utils/filter.ts:24]:
```suggestion
  if (!to) {
```

File: indexer/src/utils/filter.ts (hunk 2 above)
Comment (greged93) [reply indexer/src/utils/filter.ts:35]: I think we can remove this comment, I don't really see what more robust check we could have and also, 
any modification of the rpc codebase will lead CI to fail.

File: indexer/src/utils/filter.ts (hunk 1 above)
Comment (greged93) [reply indexer/src/utils/filter.ts:9]: https://github.com/kkrt-labs/kakarot-rpc/issues/1117

**Review by tcoratger — APPROVED** [review]

**Pull request merged**